irpc.py

```diff
@@ -64,8 +79,14 @@ def add_provider_call(funcdef: FuncDef,
     for f in l_func:
         add_provider_call(f, l_ent)
 
-    for e in l_provider:
-        hoist_declaration(ast.ext, e)
+        
+
+    for p in l_provider:
+        hoist_declaration(ast.ext, p)
+
+    for e in l_ent:
+        hoist_prov_memo(ast.ext, e)
 
+    print(ast.ext) 
```

Comment:
We don't want to print the AST

Reply:
My bad, did not mean to leave that in there. That was for testing purposes